src/root.css

```diff
@@ -10,6 +10,11 @@
   overflow-y: scroll;
 }
 
+.patientBanner {
+  position: relative;
+  z-index: 1;
```

Comment:
Using z-index should be last option @denniskigen because as the codebase grows, having a lot of z-index often leads to difficult problems to debug, imagine having 10 different items with 10 different z-indexes. Try using `position` instead.

Reply:
Noted. Here's a bunch of other ways to achieve stacking without z-index https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/Stacking_without_z-index

Reply:
What's the status of this?

Reply:
I tried using a bunch of mix of absolute and static positioning which worked in the sense that the patient profile got brought to the foreground but it also caused another problem - the NavBar was now being overlaid on top of the patient profile because it had `position: sticky` applied to it. Overall, I didn't find a way to get those two things to play well together other than z-index.

![z-index](https://user-images.githubusercontent.com/8509731/79859593-e9179480-83d9-11ea-8c27-ceb6f06a2310.png)

Reply:
:thinking:  This is a very good question -- what to do about when DOM order (which we don't have much control over in MFE) disagrees with visual stacking [order](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/Stacking_without_z-index). And probably not the last time we'll encounter it. @FlorianRappl , any guidance?

Reply:
Personally, I would avoid `z-index` as much as possible. It becomes a fight for the highest number, while still having troubles with some edge cases. That being written - sometimes it is, unfortunately, unavoidable.

Usually, the tendency would / should be not to have any `z-index` in any CSS and for cases to use specialized / "global" controls, which are either managed by the app shell, or in a dedicated MFE. There are multiple options for such controls - it does not need to be a React component.

From a component POV my only question would be: Why is there z-index etc. used anyway? Its no flyout or modal dialog here? Or maybe I miss something (sorry for the stupid question). If we can just avoid it that would be superb.

Reply:
Yeah Florian the problem is that we have in the DOM

```
<MFE 1 />
<MFE 2 />
```

and then when MFE 1 wants to launch a modal, the best it can do is

```
<MFE 1>
  <MFE 1 Content />
  <MFE 1 Modal />
</MFE 1>
<MFE 2>
```

Because of the [stacking rules](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/Stacking_without_z-index), the modal from MFE 1 winds up behind MFE 2.

Reply:
That's exactly where a central modal manager comes in handy. If MFE1 runs a modal, the modal will stay on top even if MFE2 decides (how? a modal usually comes with a blocking screen, so it usually would close - the only reason can be an event, e.g., from the backend or time-based) to open another modal dialog.

If both are not modals, but flyouts then such things should also not really happen. A flyout should still require user interaction to open and it should close when user interaction appears on another component.

Maybe my whole thinking here is academic.

Reply:
That's really valuable, thanks. I think it would be useful to try and model this as a modal or a flyout, and then build as appropriate. Which, it sounds like we should have a modal manager. @denniskigen , thoughts?

Should we accept this with a z-index for now, and then refactor once we have a modal manager?

Reply:
I think we should accept this as is, yeah @brandones.

Reply:
Sorry Dennis, that's not really the part I was asking about, of course I imagine it would be nice for you to just put it in as-is :smile: 

Rather, I'm interested in your input about the design of this feature -- is it a modal? Is it a flyout? Is it an accordion? (also, where are the designs, anyway?)

I leave it to @FlorianRappl how we should approach getting this in and making a modal manager component.

Reply:
Guilty 😬. I'll defer to @FlorianRappl as well.

Reply:
> Should we accept this with a z-index for now, and then refactor once we have a modal manager?

Yes, let's accept this for now, but then let's find out where we use `z-index` right now and why. In case of modals let's go into the direction of a modal manager. I'd do a proposal here.